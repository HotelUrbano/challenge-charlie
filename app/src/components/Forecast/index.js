import React from 'react';
import './index.css';

const Forecast = props => {
    return (
        <div>
            <div className="tomorrow">
                <h3>Amanhã</h3>
                {props.tomorrow}{' '}
                <span
                    className={
                        'unit ' +
                        (props.units.temperature === 'C'
                            ? 'celsius-unit'
                            : 'fahrenheit-unit')
                    }
                />
            </div>
            <div className="after-tomorrow">
                <h3>Depois de Amanhã</h3>
                {props.afterTomorrow}{' '}
                <span
                    className={
                        'unit ' +
                        (props.units.temperature === 'C'
                            ? 'celsius-unit'
                            : 'fahrenheit-unit')
                    }
                />
            </div>
        </div>
    );
};

export default Forecast;
