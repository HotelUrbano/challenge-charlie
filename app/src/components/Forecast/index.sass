.tomorrow
    padding: 10px
.after-tomorrow
    padding: 10px
    background-color: #0000002b