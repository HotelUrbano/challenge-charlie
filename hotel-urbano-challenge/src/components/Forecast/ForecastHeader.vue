<template>
  <md-card-header class="header">
    <div class="header__title">
      <img 
        :src="'Location' | iconWeather" 
        class="header__icon"
        alt="Ícone de Localização">
      {{ location ? location : "Não Disponível" }}
    </div>
  </md-card-header>
</template>

<script>
export default {
    name: "ForecastHeader",
    props: ["location"]
};
</script>

<style scoped lang="scss">
@import "@/styles/variables/variables.scss";
@import "@/styles/mixins/media-queries.scss";

.header {
    background: $headerColor;
    color: $headerFontColor;
    font-weight: bold;
    text-align: left;
    @include sm {
        height: initial;
    }
    .header__title {
        @include xs {
            margin-top: -8px;
        }
        font-weight: bold;
        font-size: 25px;
    }
    .header__icon {
        width: 40px;
    }
}
</style>
