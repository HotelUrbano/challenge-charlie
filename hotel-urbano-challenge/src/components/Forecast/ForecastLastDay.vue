<template>
  <div class="third-day">
    <p class="third-day__title">DEPOIS DE AMANHÃ</p>
    <div class="third-day__content">
      {{ temperature ? temperature : "N/A" }}
      <div 
        v-show="temperature" 
        class="third-day__degree">
        <p 
          :class="{inactive: unit == 'F'}"
          class="third-day__degree--celsius"
          @click="getClimate(lat, long, 'c')">ºC</p>
        <p 
          :class="{inactive: unit == 'C'}"
          class="third-day__degree--fahreheint inactive"
          @click="getClimate(lat, long, 'f')">ºF</p>
      </div>
    </div>
  </div>
</template>

<script>
import api from "@/services/api.js";

export default {
    name: "ForecastLastDay",
    props: ["unit", "temperature", "lat", "long"],
    methods: {
        getClimate(lat, long, unit) {
            return api
                .getClimate(lat, long, unit)
                .then(weather => {
                    this.$store.commit("SET_WEATHER", weather);
                })
                .catch();
        }
    }
};
</script>

<style scoped lang="scss">
@import "@/styles/mixins/media-queries.scss";
@import "@/styles/placeholders/placeholders.scss";

.third-day {
    .third-day__title {
        @extend %day;
    }
    .third-day__content {
        @extend %temperature;
    }
    .third-day__degree {
        @extend %degree;
        .third-day__degree--fahreheint {
            @extend %fahreheint;
        }
    }
    @include xs {
        margin-top: 52px;
    }
    @include md {
        margin-top: 168px;
    }
    @include lg {
        margin-top: 87px;
    }
    @include rwd(1920, 1080) {
        margin-top: 146px;
    }
}
.inactive {
    @extend %inactive;
}
</style>
