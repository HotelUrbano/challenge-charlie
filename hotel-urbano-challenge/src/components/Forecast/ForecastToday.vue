<template>
    <div class="first-day">
        <h2 class="first-day__title">HOJE</h2>
        <div class="first-day__content">
            {{temp ? temp : "N/A"}}
            <div class="first-day__degree" v-show="temp">
                <p :class="{inactive: unit == 'F'}"
                   class="first-day__degree--celsius" 
                   @click="getClimate(lat, long, 'c')">ºC</p>
                <p :class="{inactive: unit == 'C'}"
                   class="first-day__degree--fahreheint" 
                   @click="getClimate(lat, long, 'f')">ºF</p>
            </div>
        </div>
        <p class="first-day__climate">{{ condition | translate }}</p>
        <div class="first-day__contentclimate">
            <p>Vento: {{wind ? wind : "N/A"}}</p>
            <p>Humidade: {{humidity ? humidity : "N/A"}}</p>
            <p>Pressão: {{pressure ? pressure : "N/A"}}</p>
        </div>
    </div>
    
</template>

<script>
    import api from "@/services/api.js";

    export default {
        name: "ForecastToday",
        props: ['lat',
                'long',
                'temp',
                'unit',
                'condition',
                'wind',
                'humidity',
                'pressure'],
        methods: {
            getClimate(lat, long, unit) {
                return api.getClimate(lat, long, unit)
                    .then(weather => {
                        this.$store.commit("SET_WEATHER", weather);
                    })
                    .catch();
            }
        }
    }
</script>

<style scoped lang="scss">
    @import "@/styles/placeholders/placeholders.scss";

    .first-day {
        .first-day__title {
            @extend %day;
        }
        .first-day__content {
            @extend %temperature;
        }
        .first-day__degree {
            @extend %degree;
            .first-day__degree--fahreheint {
                @extend %fahreheint;
            }
            .first-day__degree--inactive {
                @extend %inactive;
            }
        }
        .first-day__climate {
            @extend %climate;
        }
        .first-day__contentclimate {
            @extend %content-climate;
        }
        .inactive {
            @extend %inactive;
        }
    }


</style>