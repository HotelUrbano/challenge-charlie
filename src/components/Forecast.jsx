import React, { Component } from 'react';
import Text from './Text';

export default class Forecast extends Component {

	constructor() {
		super();
		this.state = {
			isCelsius: true,
			today: null,
			tomorrow: null,
			afterTomorrow: null,
			temperature: null
		}
	}

	componentWillMount() {
		this.toggleTemperature(this.props.data.temperatures)
	}

	toCelsius(temperature) {
		return Math.round((parseInt(temperature) - 32) / (9 / 5));
	}
	toFarenheit(temperature) {
		return Math.round((parseInt(temperature) + 32) / (9 / 5));
	}

	toggleTemperature(temperature) {

		if(this.state.isCelsius) {
			let today = this.toCelsius(temperature[0]);	
			let tomorrow = this.toCelsius(temperature[1]);	
			let afterTomorrow = this.toCelsius(temperature[2]);	

			this.setState({
				isCelsius: false,
				today: today,
				tomorrow: tomorrow,
				afterTomorrow: afterTomorrow
			})

		} else {
			let today = this.toFarenheit(temperature[0]);	
			let tomorrow = this.toFarenheit(temperature[1]);	
			let afterTomorrow = this.toFarenheit(temperature[2]);	

			this.setState({
				isCelsius: true,
				today: today,
				tomorrow: tomorrow,
				afterTomorrow: afterTomorrow
			})
			
		}
	}
	render() {
console.log('forecast')
		return (
			<div className="forecast">
				<div className="container box-today">
					<Text 
						className="box-forecast"
						text="Hoje"/>

					<p 
						className="temperatures" 
						onClick={ this.toggleTemperature.bind(this, this.props.data.temperatures)}>
						{ this.state.isCelsius ? `${this.state.today}ºF ` : `${this.state.today}ºC ` }
					</p>
				</div>
				<div className="container box-tommorrow">
					<Text 
						className="box-forecast"
						text="Amanhã"/>
					<p 
						className="temperatures" 
						onClick={ this.toggleTemperature.bind(this, this.props.data.temperatures)}>
						{ this.state.isCelsius ? `${this.state.tomorrow}ºF ` : `${this.state.tomorrow}ºC ` }
					</p>
				</div>
				<div className="container box-after-tommorrow">
					<Text 
						className="box-forecast"
						text="Depois de amanhã"/>
					<p 
						className="temperatures" 
						onClick={ this.toggleTemperature.bind(this, this.props.data.temperatures)}>
						{ this.state.isCelsius ? `${this.state.afterTomorrow}ºF ` : `${this.state.afterTomorrow}ºC ` }
					</p>
				</div>
			</div>
		);
	}
}