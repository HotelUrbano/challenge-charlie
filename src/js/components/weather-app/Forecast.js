import React, { Component } from 'react'
import WeatherIcon from '../../utils/selectIcon'
import {convertCtoF} from '../../utils/convertCtoF'
import { connect } from 'react-redux'
import {translate} from '../../utils/translate'

class Forecast extends Component {
  render() {

    return (
      !this.props.noLocation
      ?<div className={`forecast ${this.props.colorByTemp}`}>
        <div className="left-forecast box">
          <span className="day">{this.props.lang === 'en' ? this.props.day : translate(this.props.day)}</span>
        </div>
        <div className="mid-forecast">
         {/* Seleçao de icone ... ver 'src/components/weather-icon' */}
          <WeatherIcon code={parseInt(this.props.code,10)} />
        </div>
        <div className="right-forecast box">
          <div className='temp'>
          <span><div>{`${convertCtoF(this.props.low,this.props.mainDeg)}/`}</div><div>{`${convertCtoF(this.props.high,this.props.mainDeg)}`}</div></span><span className='deg'><div className={(this.props.mainDeg === true ? 'celcius active' : 'celcius')}>°C</div><div className={(this.props.mainDeg === false ? 'fahr active' : 'celcius')}>°F</div></span>
          </div>
        </div>
        {this.props.children}
      </div>
      : <div className='forecast grey-back'></div>
      
    )
  }
}

const mapStateToProps = (state)=>{
return{
  noLocation: state.noLocation,
  mainDeg: state.mainDeg
}
}

export default connect(mapStateToProps,null)(Forecast)