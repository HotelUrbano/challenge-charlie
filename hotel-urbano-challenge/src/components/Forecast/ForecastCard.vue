<template>
  <md-card 
    v-show="!loading" 
    class="card">
    <forecast-header :location="weather.location"/>

    <md-card-content 
      id="content" 
      class="card__content">
      <img 
        :src="weather.condition | iconWeather"
        class="card__content--icon"
        alt="Ícone Meteorologia">

      <!--Conteúdo Lateral-->
      <aside class="card__content--aside">

        <!--Primeiro dia-->
        <forecast-today 
          :temp="weather.today"
          :unit="weather.unit"
          :condition="weather.condition"
          :wind="weather.wind"
          :humidity="weather.humidity"
          :pressure="weather.pressure"
          :lat="lat"
          :long="long"/>

        <!--Segundo dia-->
        <forecast-tomorrow 
          :unit="weather.unit"
          :lat="lat"
          :long="long"
          :temperature="weather.tomorrow"/>

        <!--Terceiro dia-->
        <forecast-last-day 
          :unit="weather.unit"
          :lat="lat"
          :long="long"
          :temperature="weather.after"/>
      </aside>
    </md-card-content>

  </md-card>
</template>
<script>
import ForecastHeader from "@/components/Forecast/ForecastHeader.vue";
import ForecastToday from "@/components/Forecast/ForecastToday.vue";
import ForecastTomorrow from "@/components/Forecast/ForecastTomorrow.vue";
import ForecastLastDay from "@/components/Forecast/ForecastLastDay.vue";

export default {
    name: "ForecastCard",
    components: {
        ForecastHeader,
        ForecastToday,
        ForecastTomorrow,
        ForecastLastDay
    },
    props: ["weather", "loading", "lat", "long"],
    updated() {
        this.setGradient(this.weather);
    },
    methods: {
        setGradient(weather) {
            let element = document.getElementById("content");
            if (weather.today) {
                let temp = parseInt(weather.today);
                if (weather.unit === "C" && temp < 15)
                    element.classList.add("card__content--background--blue");
                else if (weather.unit === "C" && temp > 35)
                    element.classList.add("card__content--background--red");
                else element.classList.add("card__content--background--yellow");
            } else if (!weather.today) {
                element.classList.add("card__content--background--default");
            }
        }
    }
};
</script>

<style scoped lang="scss">
@import "@/styles/variables/variables.scss";
@import "@/styles/mixins/media-queries.scss";
@import "@/styles/mixins/linear-gradient.scss";

.card {
    display: inline-block;
    vertical-align: top;
    margin-top: 30px;

    .card__content {
        @include xs {
            height: 380px;
            width: 310px;
        }
        @include md {
            height: 800px;
            width: 600px;
        }
        @include lg {
            height: 600px;
            width: 550px;
        }
        @include rwd(1920, 1080) {
            height: 800px;
            width: 600px;
        }
    }
    .card__content--background--default {
        @include linear-gradient($gradientDefault);
    }
    .card__content--background--yellow {
        @include linear-gradient($gradientYellow);
    }
    .card__content--background--blue {
        @include linear-gradient($gradientBlue);
    }
    .card__content--background--red {
        @include linear-gradient($gradientRed);
    }
    color: $contentColor;
    font-weight: bold;
    text-align: left;

    .card__content--icon {
        @include xs {
            width: 150px;
        }
        @include md {
            width: 320px;
        }
        @include lg {
            width: 245px;
        }
    }

    .card__content--aside {
        @include xs {
            left: 169px;
            top: 52px;
        }
        @include md {
            left: 340px;
        }
        @include rwd(360, 600) {
            top: 63px;
        }
        @include lg {
            left: 270px;
            top: 64px;
        }
        position: absolute;
    }
}
</style>
