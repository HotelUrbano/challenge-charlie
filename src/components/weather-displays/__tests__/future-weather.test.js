import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Forecast from "../future-weather";

describe("Forecast display", () => {
    const props = {
        temperature: 30,
        unit: "C",
        day: "Amanhã"
    };

    it("matches snapshot", () => {
        const component = renderer.create(<Forecast {...props} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("shows correct information", () => {
        const wrapper = mount(<Forecast {...props} />);

        expect(wrapper.find("h2").text()).toBe("Amanhã");
        expect(wrapper.find("span.temperature").text()).toBe("30 ºC");
    });
});
