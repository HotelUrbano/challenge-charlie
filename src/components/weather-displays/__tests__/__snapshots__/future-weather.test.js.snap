// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast display matches snapshot 1`] = `
<div
  className="sc-bdVaJa eYGOmH"
  color="#FFEB3Bff"
>
  <div />
  <div>
    <h2>
      Amanhã
    </h2>
    <p>
      <span
        className="temperature"
      >
        30
         
        º
        C
      </span>
    </p>
  </div>
</div>
`;
